eZ/Bundle/EzPublishCoreBundle/Command/RegenerateUrlAliasesCommand.php

```diff
@@ -0,0 +1,238 @@
+                $locationsCount,
+                self::BEFORE_RUNNING_HINTS
+            ),
+            false
+        );
+        if (!$helper->ask($input, $output, $question)) {
+            return;
+        }
+
+        $output->writeln('Regenerating System URL aliases...');
+
+        $progressBar = $this->getProgressBar($locationsCount, $output);
+        $progressBar->start();
+
+        for ($offset = 0; $offset <= $locationsCount; $offset += $iterationCount) {
+            gc_disable();
+            $locations = $this->repository->sudo(
+                function (Repository $repository) use ($offset, $iterationCount) {
+                    return $repository->getLocationService()->loadAllLocations($offset, $iterationCount);
+                }
+            );
+            $this->processLocations($locations, $progressBar);
+            gc_enable();
+        }
+        $progressBar->finish();
+        $output->writeln('');
+
+        $output->writeln('<info>Cleaning up corrupted URL aliases...</info>');
+        $corruptedAliasesCount = $this->repository->sudo(
+            function (Repository $repository) {
+                return $repository->getURLAliasService()->deleteCorruptedUrlAliases();
+            }
+        );
+        $output->writeln("<info>Done. Deleted {$corruptedAliasesCount} entries.</info>");
+        $output->writeln('Purging HTTP cache...');
+        $this->cachePurger->purgeAll();
+        $output->writeln('<info>Done.</info>');
+    }
+
+    /**
+     * Return configured progress bar helper.
+     *
+     * @param int $maxSteps
+     * @param \Symfony\Component\Console\Output\OutputInterface $output
+     *
+     * @return \Symfony\Component\Console\Helper\ProgressBar
+     */
+    protected function getProgressBar($maxSteps, OutputInterface $output)
+    {
+        $progressBar = new ProgressBar($output, $maxSteps);
+        $progressBar->setFormat(
+            ' %current%/%max% [%bar%] %percent:3s%% %elapsed:6s%/%estimated:-6s% %memory:6s%'
+        );
+
+        return $progressBar;
+    }
+
+    /**
+     * Process single results page of fetched Locations.
+     *
+     * @param \eZ\Publish\API\Repository\Values\Content\Location[] $locations
+     * @param \Symfony\Component\Console\Helper\ProgressBar $progressBar
+     */
+    private function processLocations(array $locations, ProgressBar $progressBar)
+    {
+        $contentList = $this->repository->sudo(
+            function (Repository $repository) use ($locations) {
+                $contentInfoList = array_map(
+                    function (Location $location) {
+                        return $location->contentInfo;
+                    },
+                    $locations
+                );
+
+                // load Content list in all languages
+                return $repository->getContentService()->loadContentListByContentInfo(
+                    $contentInfoList,
+                    [],
```

Comment:
You can safely backport `Language::ALL` _(constant only)_ so this as well as other people's code can start to get forward compatible with 3.x _(or whenever we switch siteaccess awareness on by default)_.

ref: https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/API/Repository/Values/Content/Language.php#L23-L26

Reply:
Added via fixup: de332cf